cmd/akash/provider.go

```diff
@@ -121,11 +121,12 @@ func runCommand() *cobra.Command {
 		Use:   "run <provider>",
 		Short: "respond to chain events",
 		Args:  cobra.ExactArgs(1),
-		RunE:  session.WithSession(session.RequireNode(doProviderRunCommand)),
+		RunE:  session.WithSession(session.RequireNode(session.RequireHost(doProviderRunCommand))),
 	}
 
 	cmd.Flags().Bool("kube", false, "use kubernetes cluster")
 	cmd.Flags().String("manifest-ns", "lease", "set manifest namespace")
+	cmd.Flags().String("host", "", "cluster host")
```

Comment:
make a `session.AddFlagHost()` and do this there.  also use `viper.BindPFlag()` instead of `viper.BindEnv()` in it.